### ISSUE TITLE
[BUG] Alias with malformed declaration leads to Segfault

### ISSUE THREAD

**Issue by realgdman** [hsutter/cppfront#355]
**Describe the bug**
I accidentally wrote malformed alias (guess it's unnamed temporary variable?) which leads to Segfault.

**Minimal reproduce**
`alias1: == std::function<(:int)>;`
And after some experiments I think same reason as
`alias2: == (:int);`

**Command lines including which C++ compiler you are using**
`cppfront example.cpp2 -p`
(build from latest mai. f83ca9b with clang++-15) 

**Expected result - what you expected to happen**
Some diagnostic message. Note that 
`alias3: == a:int;` gives diagnostic `error: ';' expected at end of alias declaration`
`alias4: == (a:int);` gives diagnostic `error: unexpected text - expression-list is not terminated by ) (at ':')`
`error: an object '==' alias declaration must be followed by an expression`

**Actual result/error**
**Additional context**
I've done some limited debugging, it hits `assert(decl->initializer)` at https://github.com/hsutter/cppfront/blob/f83ca9b727e9c07f60a1933584a49a3d971936f5/source/cppfront.cpp#L2385
Which is called from https://github.com/hsutter/cppfront/blob/f83ca9b727e9c07f60a1933584a49a3d971936f5/source/cppfront.cpp#L2356
https://github.com/hsutter/cppfront/blob/f83ca9b727e9c07f60a1933584a49a3d971936f5/source/cppfront.cpp#L1573
>//  Handle object aliases
https://github.com/hsutter/cppfront/blob/f83ca9b727e9c07f60a1933584a49a3d971936f5/source/cppfront.cpp#L4738

**Comment by filipsajdak** [comment]
You are trying to define a `std::function` using cpp2 function declaration - it uses an unnamed argument, which is not yet supported. See: #343

**Comment by realgdman** [comment]
I agree about `std::function` parameter not implemented, but please look at this example (granted it may be meaningless):
`main: () -> int = {`
`	(:int);`
`}`
It segfaults at the same line 2385
